packages/b/bgfx/rules/shaders.lua

```diff
@@ -24,7 +25,13 @@ rule("shaders")
         if fileconfig and fileconfig.output_name then
             output_filename = fileconfig.output_name
         else
-            output_filename = path.filename(shaderfile) .. ".bin"
+            local filename = path.filename(shaderfile)
+            output_filename = filename:match("^(.*)%.sc$") or filename
```

Comment:
`path.basename` can also remove the extension. Btw, is it necessary to remove the .sc extension? when there is shader.vert  and shader.vert.sc, it can cause conflicts.

Reply:
I removed the ".sc" extension because it's the source code extension (extensively used in the project sources), but only if it is set, also that's why I used the match function instead of basename, because this way I can remove the ".sc" extension only if it's used. If we have the situation you mentioned both will generate the same output file, indeed. The transformation I used follows the bgfx example files, and the output_name can be used anyway.